exmg-sortable.ts

```diff
@@ -0,0 +1,395 @@
+  private animationPromise?: Promise<any>;
+
+  constructor() {
+    super();
+
+    /* Save function references */
+    this.handleTrack = this.handleTrack.bind(this);
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+    addListener(this, 'track', this.handleTrack);
+  }
+
+  disconnectedCallback(): void {
+    removeListener(this, 'track', this.handleTrack);
+  }
+
+  /**
+   * Tracks a pointer from touchstart/mousedown to touchend/mouseup. Note that the start state is fired following
+   * the first actual move event following a touchstart/mousedown.
+   */
+  private handleTrack(e:Event): void {
+    switch ((<CustomEvent>e).detail.state) {
+      case 'start':
+        if (!this.dragRequestPending) {
+          this.trackStart(e);
+        }
+        break;
+      case 'track':
+        this.trackMove(e);
+        break;
+      case 'end':
+        if (this.animationPromise) {
+          this.animationPromise.then(() => {
+            afterNextRender(this, this.trackEnd);
+          });
+        } else {
+          afterNextRender(this, this.trackEnd);
+        }
+        break;
+    }
+  }
+
+  /**
+   * Initialized a drag and drop sequence if a child node was clicked that matches the itemSelector property. If a
+   * handleSelector is defined, a node matching this selector must be clicked instead.
+   */
+  private trackStart(e:Event): void {
+    const handle = this.handleSelector;
+    const eventPath: EventTarget[] = (<any>e).path ? (<any>e).path : e.composedPath();
+    const targetElement = <HTMLElement>eventPath[0];
+
+    /* Look for closest handle */
+    if (handle && !targetElement.closest(handle)) {
+      return;
+    }
+
+    this.style.webkitUserSelect = 'none';
+    this.dragRequestPending = true;
+
+    const selector = this.itemSelector;
+    const node = <HTMLElement>targetElement.closest(selector);
+
+    if (node) {
+      e.preventDefault();
+
+      this.draggedElement = node;
+      this.sortableNodes = Array.from(this.querySelectorAll(selector)) || [];
+      this.draggedElementClone = this.createClone(node);
+      this.draggedElementOrigin = <HTMLElement>node.nextSibling;
+      this.animatedElements = [];
+
+      this.draggedElement!.classList.add(this.draggedClass);
+    }
+  }
+
+  /**
+   * Ends the drag and drop sequence.
+   */
+  private trackEnd(): void {
+    if (!this.draggedElement) {
+      return;
+    }
+
+    const updated = Array.from(this.querySelectorAll(this.itemSelector));
+
+    const sourceIndex = this.sortableNodes.indexOf(this.draggedElement);
+    const targetIndex = updated.indexOf(this.draggedElement);
+
+    if (sourceIndex !== targetIndex) {
+      this.dispatchEvent(new CustomEvent('dom-order-change', {
+        bubbles: true,
+        composed: true,
+        detail: {
+          sourceIndex,
+          targetIndex,
+        },
+      }));
+    }
+
+    this.reset();
+
+    this.dragRequestPending = false;
+    this.style.webkitUserSelect = 'text';
```

Comment:
You only updated webkitUserSelect what about other browsers ? Shouldn't you also update userSelect property ?

Reply:
done